### PR TITLE
feat(__init__): accept api key as a string or from environment

### PR DESCRIPTION
**Pull request by devngc**

### CODE REVIEW

File: pollination_apps/cli/__init__.py [hunk 1]
```diff
@@ -84,21 +83,37 @@ def login(environment: str, token_name: str):
...
+    default='production', show_default=True
 )
 @click.option(
     '--public/--private', help='Indicate if the application should be created as '
     'a public or a private resource. This option does not change the visibility of a '
-    'resource if it already exist.', is_flag=True, default=True
+    'resource if it already exist.', is_flag=True, default=True, show_default=True
+)
+@click.option(
+    '-api', '--api-token', help='valid Pollination api token', default=None,
```
Comment (mostaphaRoudsari) [inline pollination_apps/cli/__init__.py:94]:
```suggestion
    '-api', '--api-token', help='A valid Pollination API token', default=None,
```

File: pollination_apps/cli/__init__.py [hunk 2]
```diff
@@ -84,21 +83,37 @@ def login(environment: str, token_name: str):
...
+    show_default=True
 )
-def deploy(path, owner, name, tag, message, environment, public):
+@click.option(
+    '-afe', '--api-from-environment', is_flag=True,
+    help='use the api token from the environment', default=False,
+    show_default=True
+)
```
Comment (mostaphaRoudsari) [inline pollination_apps/cli/__init__.py:101]: We don't need this. You can do try/except. By default, the order should be:

1. Try to read from the CLI input
2. Try to get it from the environmental variable
3. Raise an error

**Review by mostaphaRoudsari — CHANGES_REQUESTED** [review]
Hi @devngc! See my comments.

Also, as I mentioned on Basecamp, we need to update the README file in the template.

**Comment by devngc** [comment]
For the README file I created an [issue](https://github.com/pollination/pollination-apps/issues/6) and a [basecamp task](https://3.basecamp.com/4298486/buckets/27520431/todos/4945637455) as well.

**Review by mostaphaRoudsari — APPROVED** [review]
LGTM!

**Pull request merged**